cmd/state-installer/test/integration/installer_int_test.go

```diff
@@ -333,18 +328,27 @@ func (suite *InstallerIntegrationTestSuite) AssertConfig(ts *e2e.Session) {
 	}
 }
 
-func (s *InstallerIntegrationTestSuite) setupTest(ts *e2e.Session) {
+func installationDir(ts *e2e.Session) string {
+	return filepath.Join(ts.Dirs.Work, "installation")
+}
+
+type payload struct {
+	installer string
+}
+
+func (suite *InstallerIntegrationTestSuite) newPayload(ts *e2e.Session) *payload {
 	root := environment.GetRootPathUnsafe()
-	buildDir := fileutils.Join(root, "build")
-	installerExe := filepath.Join(buildDir, constants.StateInstallerCmd+osutils.ExeExt)
-	if !fileutils.FileExists(installerExe) {
-		s.T().Fatal("E2E tests require a state-installer binary. Run `state run build-installer`.")
-	}
-	s.installerExe = ts.CopyExeToDir(installerExe, filepath.Join(ts.Dirs.Base, "installer"))
+	buildDir := filepath.Join(root, "build")
+	payDir := filepath.Join(ts.Dirs.Work, "payload")
+
+	suite.Assert().NoError(fileutils.MkdirUnlessExists(payDir))
 
-	payloadDir := filepath.Dir(s.installerExe)
-	ts.CopyExeToDir(ts.Exe, filepath.Join(payloadDir, installation.BinDirName))
-	ts.CopyExeToDir(ts.SvcExe, filepath.Join(payloadDir, installation.BinDirName))
+	err := paygen.GeneratePayload(buildDir, payDir)
+	suite.Assert().NoError(err, "generate payload")
```

Comment:
I've been thinking about this more and I'm just not a big fan of this. All our current integration tests require the bits to be there, they do not generate bits, just test them. So there's the consistency factor, but there's also the fact that this will redundantly generate a payload each run, and will override whatever I may have generated manually.

I'd much rather we keep this the same way all our other integration tests run: The user (us) must manually run `generate-payload`, which happens on a CI step for automations. The test should fail if the payload dir does not exist or is empty.